direct/algorithms/mri.py

```diff
@@ -0,0 +1,179 @@
+# coding=utf-8
+# Copyright (c) DIRECT Contributors
+
+from typing import Any, Callable, Dict
+
+import numpy as np
+import torch
+from torch import nn
+
+from direct.algorithms.optimization import MaximumEigenvaluePowerMethod
+from direct.data.transforms import view_as_complex, view_as_real
+from direct.utils import DirectTransform
+
+
+class EspiritCalibration(DirectTransform):
+    """Estimates sensitivity maps estimated with the ESPIRIT calibration method as described in [1]_.
+
+    We adapted code for ESPIRIT method adapted from [2]_.
+
+    References
+    ----------
+
+    .. [1] Uecker M, Lai P, Murphy MJ, Virtue P, Elad M, Pauly JM, Vasanawala SS, Lustig M. ESPIRiT--an eigenvalue
+        approach to autocalibrating parallel MRI: where SENSE meets GRAPPA. Magn Reson Med. 2014 Mar;71(3):990-1001.
+        doi: 10.1002/mrm.24751. PMID: 23649942; PMCID: PMC4142121.
+    .. [2] https://github.com/mikgroup/sigpy/blob/1817ff849d34d7cbbbcb503a1b310e7d8f95c242/sigpy/mri/app.py#L388-L491
+
+    """
+
+    def __init__(
+        self,
+        backward_operator: Callable,
+        threshold: float = 0.05,
+        kernel_size: int = 6,
+        crop: float = 0.95,
+        max_iter: int = 100,
+        kspace_key: str = "masked_kspace",
+    ):
+        """Inits :class:`EstimateSensitivityMap`.
```

Comment:
? isn't this a different class

Reply:
Can't kspace_key be an Enum, just like https://github.com/kornia/kornia/blob/b4ab100a1d2b87f8ae290dbd227c407c06f1d3d3/kornia/constants.py#L125

Reply:
Done